README.md

```diff
@@ -68,49 +68,81 @@ trigger OppRollup on Opportunity (after insert, after update,
             new LREngine.RollupSummaryField(
                                             Schema.SObjectType.Account.fields.AnnualRevenue,
                                             Schema.SObjectType.Opportunity.fields.Amount,
-                                            LREngine.RollupOperation.Sum 
-                                         )); 
+                                            LREngine.RollupOperation.Sum
+                                         ));
      ctx.add(
             new LREngine.RollupSummaryField(
                                             Schema.SObjectType.Account.fields.SLAExpirationDate__c,
                                                Schema.SObjectType.Opportunity.fields.CloseDate,
                                                LREngine.RollupOperation.Max
-                                         ));                                       
-	 
-     /* 
-      Calling rollup method returns in memory master objects with aggregated values in them. 
-      Please note these master records are not persisted back, so that client gets a chance 
+                                         ));                                      
+  
+      /*
+        To Prepare MasterId's incase if the realtionship is changed on child object Old Master record and nNew Master Records
+        needs recalculation of Rollup Summaries.
+      */
+
+      set<id> MasterIds=new set<id>();
+
+        for( Opportunity newOpportunity:objects){
+            if(trigger.isupdate){
+                Opportunity oldOpportunity=(Opportunity)trigger.oldMap.get(newOpportunity.Id);
+                if(oldOpportunity!=null&& newOpportunity.AccountId!=oldContact.AccountId){
+                      MasterIds.add(oldOpportunity.AccountId);
+                      MasterIds.add(newOpportunity.AccountId);
+                      }
+            }else{
+                MasterIds.add(newOpportunity.AccountID);
+            }
+        }
+
+      /*
+      Calling rollup method returns in memory master objects with aggregated values in them.
+      Please note these master records are not persisted back, so that client gets a chance
       to post process them after rollup
       */ 
-     Sobject[] masters = LREngine.rollUp(ctx, objects);    
-
+  Sobject[] masters = LREngine.rollUp(ctx, MasterIds);
+
+    /*
+      To check if the Masterid's were returned to perform update operation.
+      To avoid error when child records are created with out master like Tasks, Events with out selecting Account.
```

Comment:
To avoid error when child records are created with out master like Tasks, Events with out selecting Master Relation.